packages/next/src/server/app-render/dynamic-rendering.ts

```diff
@@ -30,12 +30,39 @@ import { getPathname } from '../../lib/url'
 
 const hasPostpone = typeof React.unstable_postpone === 'function'
 
-// Stores dynamic reasons used during a render
-export type PrerenderState = { hasDynamic: boolean }
+type DynamicAccess = {
+  /**
+   * If debugging, this will contain the stack trace of where the dynamic access
+   * occurred. This is used to provide more information to the user about why
+   * their page is being rendered dynamically.
+   */
+  stack?: string
 
-export function createPrerenderState(): PrerenderState {
+  /**
+   * The expression that was accessed dynamically.
+   */
+  expression: string
+}
+
+// Stores dynamic reasons used during a render.
+export type PrerenderState = {
+  /**
+   * When true, stack information will also be tracked during dynamic access.
+   */
+  isDebugSkeleton: boolean | undefined
```

Comment:
this isn't really a property of the prerenderState but a property of the render itself. this object should generally hold mutable state about the the render and isDebugSkeleton shouldn't be mutated by the render itself so it shouldn't really live here